xayn-ai/src/data/document.rs

```diff
@@ -15,7 +15,7 @@ use super::document_data::DocumentDataWithMab;
 
 #[repr(transparent)]
 #[cfg_attr(test, derive(Default))]
-#[derive(Debug, PartialEq, Eq, Clone, Hash, Serialize, Deserialize, Display)]
+#[derive(Debug, PartialEq, Eq, Clone, Copy, Hash, Serialize, Deserialize, Display)]
```

Comment:
Is there a reason to constraint this to `Copy`? It remove some clone but we are going to copy always 128bit instead of the smaller reference.

Reply:
It should still be faster to copy a 128bit value then to use a reference to a 128bit value, in general.

The few (64) additional bites copied on the stack normally do not out-weight the benefits from not
needing to de-reference the value when using it, even if it's mostly copied around.

There was a good article about when to pass by-value and when to pass by reference a few years ago,
let me see if I can find it.

If I remember correctly it was something like:

- On 64bit passing 128bit by value can be either faster or similar fast depending on circumstance.

- On 32bit passing 128bit by value might be slightly slower, but you need to try really hard to measure it as in most contexts it is such a small difference that it is drowned by noise.

I mean the only way passing a 128bit value is potentially slower (in this use-case) is if:

- You could pass it to function in the register instead of using the stack.
- and we pass it through a bunch of function calls, through register before using it.

E.g. if we have `foobar(&uuid)` and `foobar` is a leaf function we do pass the `&` 
most likely in registers (at least on 32bit arm EABI) but then we de-reference it 
(from the stack), making it maybe even (neglible) slower then just directly passing
it through the stack from the get to go. 

And on 64bit we can pass uuid directly through in registers (normally).

Sure things are more complex when multiple argument are involved etc.

But then the stack is (normally) in the L1 D-Cache so it's not that much slower.

And non of this code is in tight/hot loop.

Because of this I normally always pass UUIDs by Copy without thinking about it much,
I double we could measure a perf. difference between `&DocumentId` and `DocumentId` 
in our bussiness-use-case (reranking).

And `Copy` is often simpler to use.

Reply:
So should I undo the change? Or is it ok to keep it?

Reply:
Keep it

Reply:
Sorry I completely lost this comment.